vumi/components/message_store_migrators.py

```diff
@@ -53,17 +53,21 @@ def migrate_from_1(self, mdata):
         mdata.set_value('batches', mdata.old_data.get('batches', []))
         mdata.copy_values('message')
         mdata.copy_indexes('message_bin')
+        mdata.copy_indexes('message_with_status_bin')
 
         return mdata
 
     def reverse_from_2(self, mdata):
-        # We copy the `batches` field even though the older model version
-        # doesn't know about it. This lets us migrate v2 -> v1 -> v2 without
-        # losing data.
+        # We copy the `batches` field and related indexs even though the older
+        # model version doesn't know about them. This lets us migrate
+        # v2 -> v1 -> v2 without losing data.
```

Comment:
The batch information can only come from outside the migration function, so throwing it away on a reverse migration is problematic.